=== modified file 'Onboard/osk/osk_virtkey.c'
--- Onboard/osk/osk_virtkey.c	2016-10-21 23:45:34 +0000
+++ Onboard/osk/osk_virtkey.c	2020-06-19 12:00:55 +0000
@@ -936,142 +936,142 @@
     char* label;
 
     switch (keysym) {
-        case GDK_KEY_Scroll_Lock:
-            label = "Scroll\nLock";
-            break;
-
-        case GDK_KEY_space:
-            label = " ";
-            break;
-
-        case GDK_KEY_Sys_Req:
-            label = "Sys Rq";
-            break;
-
-        case GDK_KEY_Page_Up:
-            label = "Page\nUp";
-            break;
-
-        case GDK_KEY_Page_Down:
-            label = "Page\nDown";
-            break;
-
-        case GDK_KEY_Num_Lock:
-            label = "Num\nLock";
-            break;
-
-        case GDK_KEY_KP_Page_Up:
-            label = "Pg Up";
-            break;
-
-        case GDK_KEY_KP_Page_Down:
-            label = "Pg Dn";
-            break;
-
-        case GDK_KEY_KP_Home:
-            label = "Home";
-            break;
-
-        case GDK_KEY_KP_Left:
-            label = "Left";
-            break;
-
-        case GDK_KEY_KP_End:
-            label = "End";
-            break;
-
-        case GDK_KEY_KP_Up:
-            label = "Up";
-            break;
-
-        case GDK_KEY_KP_Begin:
-            label = "Begin";
-            break;
-
-        case GDK_KEY_KP_Right:
-            label = "Right";
-            break;
-
-        case GDK_KEY_KP_Enter:
-            label = "Enter";
-            break;
-
-        case GDK_KEY_KP_Down:
-            label = "Down";
-            break;
-
-        case GDK_KEY_KP_Insert:
-            label = "Ins";
-            break;
-
-        case GDK_KEY_KP_Delete:
-            label = "Del";
-            break;
-
-        case GDK_KEY_dead_grave:
-            label = "ˋ";
-            break;
-
-        case GDK_KEY_dead_acute:
-            label = "ˊ";
-            break;
-
-        case GDK_KEY_dead_circumflex:
-            label = "ˆ";
-            break;
-
-        case GDK_KEY_dead_tilde:
-            label = "~";
-            break;
-
-        case GDK_KEY_dead_macron:
-            label = "ˉ";
-            break;
-
-        case GDK_KEY_dead_breve:
-            label = "˘";
-            break;
-
-        case GDK_KEY_dead_abovedot:
-            label = "˙";
-            break;
-
-        case GDK_KEY_dead_diaeresis:
-            label = "¨";
-            break;
-
-        case GDK_KEY_dead_abovering:
-            label = "˚";
-            break;
-
-        case GDK_KEY_dead_doubleacute:
-            label = "˝";
-            break;
-
-        case GDK_KEY_dead_caron:
-            label = "ˇ";
-            break;
-
-        case GDK_KEY_dead_cedilla:
-            label = "¸";
-            break;
-
-        case GDK_KEY_dead_ogonek:
-            label = "˛";
-            break;
-
-        case GDK_KEY_dead_belowdot:
-            label = ".";
-            break;
-
-        case GDK_KEY_Mode_switch:
-            label = "AltGr";
-            break;
-
-        case GDK_KEY_Multi_key:
-            label = "Compose";
-            break;
-
+//        case GDK_KEY_Scroll_Lock:
+//            label = "Scroll\nLock";
+//            break;
+//
+//        case GDK_KEY_space:
+//            label = " ";
+//            break;
+//
+//        case GDK_KEY_Sys_Req:
+//            label = "Sys Rq";
+//            break;
+//
+//        case GDK_KEY_Page_Up:
+//            label = "Page\nUp";
+//            break;
+//
+//        case GDK_KEY_Page_Down:
+//            label = "Page\nDown";
+//            break;
+//
+//        case GDK_KEY_Num_Lock:
+//            label = "Num\nLock";
+//            break;
+//
+//        case GDK_KEY_KP_Page_Up:
+//            label = "Pg Up";
+//            break;
+//
+//        case GDK_KEY_KP_Page_Down:
+//            label = "Pg Dn";
+//            break;
+//
+//        case GDK_KEY_KP_Home:
+//            label = "Home";
+//            break;
+//
+//        case GDK_KEY_KP_Left:
+//            label = "Left";
+//            break;
+//
+//        case GDK_KEY_KP_End:
+//            label = "End";
+//            break;
+//
+//        case GDK_KEY_KP_Up:
+//            label = "Up";
+//            break;
+//
+//        case GDK_KEY_KP_Begin:
+//            label = "Begin";
+//            break;
+//
+//        case GDK_KEY_KP_Right:
+//            label = "Right";
+//            break;
+//
+//        case GDK_KEY_KP_Enter:
+//            label = "Enter";
+//            break;
+//
+//        case GDK_KEY_KP_Down:
+//            label = "Down";
+//            break;
+//
+//        case GDK_KEY_KP_Insert:
+//            label = "Ins";
+//            break;
+//
+//        case GDK_KEY_KP_Delete:
+//            label = "Del";
+//            break;
+//
+//        case GDK_KEY_dead_grave:
+//            label = "ˋ";
+//            break;
+//
+//        case GDK_KEY_dead_acute:
+//            label = "ˊ";
+//            break;
+//
+//        case GDK_KEY_dead_circumflex:
+//            label = "ˆ";
+//            break;
+//
+//        case GDK_KEY_dead_tilde:
+//            label = "~";
+//            break;
+//
+//        case GDK_KEY_dead_macron:
+//            label = "ˉ";
+//            break;
+//
+//        case GDK_KEY_dead_breve:
+//            label = "˘";
+//            break;
+//
+//        case GDK_KEY_dead_abovedot:
+//            label = "˙";
+//            break;
+//
+//        case GDK_KEY_dead_diaeresis:
+//            label = "¨";
+//            break;
+//
+//        case GDK_KEY_dead_abovering:
+//            label = "˚";
+//            break;
+//
+//        case GDK_KEY_dead_doubleacute:
+//            label = "˝";
+//            break;
+//
+//        case GDK_KEY_dead_caron:
+//            label = "ˇ";
+//            break;
+//
+//        case GDK_KEY_dead_cedilla:
+//            label = "¸";
+//            break;
+//
+//        case GDK_KEY_dead_ogonek:
+//            label = "˛";
+//            break;
+//
+//        case GDK_KEY_dead_belowdot:
+//            label = ".";
+//            break;
+//
+//        case GDK_KEY_Mode_switch:
+//            label = "AltGr";
+//            break;
+//
+//        case GDK_KEY_Multi_key:
+//            label = "Compose";
+//            break;
+//
         default:
         {
             static char buf[256];
@@ -1104,7 +1104,7 @@
                     }
                     else
                     {
-                        buf[MIN(l, 2)] = '\0';
+//                        buf[MIN(l, 2)] = '\0';
                     }
                     label = buf;
                 }

